python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
+        ...                        IndexedRow(1, Vectors.dense([4, 5, 6])),
+        ...                        IndexedRow(2, Vectors.dense([7, 8, 9])),
+        ...                        IndexedRow(3, Vectors.dense([10, 11, 12]))])
+        >>> rm = IndexedRowMatrix(rows, 7, 6)
+        >>> int(rm.numRows())
+        7
+        """
+        return self._jirm.call("numRows")
+
+    def numCols(self):
+        """
+        Get or compute the number of cols.
+
+        >>> rows = sc.parallelize([IndexedRow(0, Vectors.dense([1, 2, 3])),
+        ...                        IndexedRow(1, Vectors.dense([4, 5, 6])),
+        ...                        IndexedRow(2, Vectors.dense([7, 8, 9])),
+        ...                        IndexedRow(3, Vectors.dense([10, 11, 12]))])
+        >>> rm = IndexedRowMatrix(rows)
+        >>> int(rm.numCols())
+        3
+
+        >>> rows = sc.parallelize([IndexedRow(0, Vectors.dense([1, 2, 3])),
+        ...                        IndexedRow(1, Vectors.dense([4, 5, 6])),
+        ...                        IndexedRow(2, Vectors.dense([7, 8, 9])),
+        ...                        IndexedRow(3, Vectors.dense([10, 11, 12]))])
+        >>> rm = IndexedRowMatrix(rows, 7, 6)
+        >>> int(rm.numCols())
+        6
+        """
+        return self._jirm.call("numCols")
+
+    def toRowMatrix(self):
+        """
+        Convert this matrix to a RowMatrix.
+
+        >>> rows = sc.parallelize([IndexedRow(0, Vectors.dense([1, 2, 3])),
+        ...                        IndexedRow(6, Vectors.dense([4, 5, 6]))])
+        >>> rm = IndexedRowMatrix(rows).toRowMatrix()
+        >>> rm.rows.collect()
+        [DenseVector([1.0, 2.0, 3.0]), DenseVector([4.0, 5.0, 6.0])]
+        """
+        javaRowMatrix = self._jirm.call("toRowMatrix")
+        return RowMatrix._from_java(javaRowMatrix)
+
+    def toCoordinateMatrix(self):
+        """
+        Convert this matrix to a CoordinateMatrix.
+
+        >>> rows = sc.parallelize([IndexedRow(0, Vectors.dense([1, 0])),
+        ...                        IndexedRow(6, Vectors.dense([0, 5]))])
+        >>> cm = IndexedRowMatrix(rows).toCoordinateMatrix()
+        >>> cm.entries.take(3)
+        [MatrixEntry(0, 0, 1.0), MatrixEntry(0, 1, 0.0), MatrixEntry(6, 0, 0.0)]
+        """
+        javaCoordinateMatrix = self._jirm.call("toCoordinateMatrix")
+        return CoordinateMatrix._from_java(javaCoordinateMatrix)
+
+
+class MatrixEntry(object):
+    """
+    Represents an entry of a CoordinateMatrix.
+
+    Just a wrapper over a (long, long, float) tuple.
+
+    .. note:: Experimental
+    """
+    def __init__(self, i, j, value):
+        self.i = long(i)
+        self.j = long(j)
+        self.value = float(value)
+
+    def __repr__(self):
+        return "MatrixEntry(%s, %s, %s)" % (self.i, self.j, self.value)
+
+
+class CoordinateMatrix(object):
+    """
+    Represents a matrix in coordinate format.
+
+    .. note:: Experimental
```

Comment:
It looks like the convention is to move the Experimental note to the top of the file.